### PR TITLE
Kraken/text input v7

### PR DESCRIPTION
**Pull request by tesk9**
New version of TextInput that adds first class support for:
- Password input show/hide PW functionality
- Guidance that can be attached to the input when the error isn't showing

## For reviewer

Instead of looking at the whole diff, I recommend ignoring the first commit (see https://github.com/NoRedInk/noredink-ui/pull/759/files/f00b1a8e9b935eae76b46787d0802c44e6b8b217..c9da21b1cd1bb6901531ab4f5d10482ede096943). This will give you a much cleaner (although still large) view!

## Visual changes

(Note: I renamed `password` to `newPassword` in the styleguide -- see deploy preview for most-up-to-date version)

**With no guidance, no error message, and no chars**

**With guidance, no error message, and no chars**

**With guidance, error message, and no chars**

**With guidance, no error message, and chars**

**With guidance, no error message, chars, and user hit "Show password"**

cc @NoRedInk/design 

## Accessibility notes

I looked around a bit online for instructions on show/hide password behavior. It seems like often folks use a checkbox or a "toggle" button. Both of these options seems strange to me since we're changing the text of the button from "Show" to "Hide".

Would love to talk through some options at some point! (I don't think this is blocking -- we should be able to merge & release and continue to iterate on the underlying implementation without changing the API) cc @krinhorn

### CODE REVIEW

**Comment by bendansby** [comment]
Love it!

defaultMarginTop should be 5

**Comment by tesk9** [comment]
Sounds good! I think I'd prefer to change that in a separate release, because I know I did some sneaky shifting by pixel value in a couple places in the monolith (specifically related to the search input needing to be vertically aligned with other content)

**Comment by bendansby** [comment]
Can we add padding to avoid password manager collision like we do for "Forgot?"

image

**Comment by ap-nri** [comment]
I can't review this right now because I'm on my phone but this rocks and you rock for making it. I am thrilled. I am a being of pure joy.

**Comment by tesk9** [comment]
Okay I looked at the "Forgot?" styles and it seems like the floating button is positioned the same distance from the righthand side. LastPass positions its little icon thing as a background with `background-position: 98% 50%`... I think the password managers will end up looking about the same in context. Although we could try setting a different background position with `important`?

`right: 15px` is used in both this component and on /log in:
Screen Shot 2021-11-01 at 10 10 45 AM
Screen Shot 2021-11-01 at 10 10 11 AM


Somewhat separately, I'm a little worried that `Show password` takes up too much room when the viewport or container is small, like in the log in header dropdown: 

image

**Comment by bendansby** [comment]
You noted here that LastPass behaves differently from the other password managers; https://github.com/NoRedInk/NoRedInk/pull/35158#pullrequestreview-748301845

**Comment by tesk9** [comment]
Screen Shot 2021-11-01 at 2 12 25 PM


Screen Shot 2021-11-01 at 2 12 31 PM

**Comment by bendansby** [comment]
Thanks! I think the narrow input concern should be fine since even in the login dropdown it's wider than that, but we can see!

**Comment by tesk9** [comment]
Phew! I'm glad adding padding worked for 1password! I was confused for a while on this, thinking that you were asking me to shift the "Show PW" message further left, further squishening the visible characters. Thanks for sticking with me on this!

**Review by ap-nri — CHANGES_REQUESTED** [review]
one question

File: src/Nri/Ui/TextInput/V7.elm [hunk 1]
```diff
@@ -402,7 +402,7 @@ type alias Config msg =
     , noMarginTop : Bool
     , css : List (List Css.Style)
     , id : Maybe String
-    , custom : List (Html.Attribute msg)
+    , custom : List (Html.Attribute Never)
```
[the remark this answers is not in the document]
Comment (ap-nri) [reply src/Nri/Ui/TextInput/V7.elm:474]: if so, then ready to FIRE AWAY

**Comment by ap-nri** [comment]
I like the way this works out on Safari 15.1
image

File: src/Nri/Ui/TextInput/V7.elm (hunk 1 above)
[the remark this answers is not in the document]
Comment (tesk9) [reply src/Nri/Ui/TextInput/V7.elm:474]: No, it wouldn't. I will add an onFocus attribute though!

[the remark this answers is not in the document]
Comment (tesk9) [reply src/Nri/Ui/TextInput/V7.elm:484]: Done 😄

**Comment by ap-nri** [comment]
@tesk9 I'm pro merging now!! If you are too then here's your gif

**Pull request merged**